batch_requests.go

```diff
@@ -210,13 +211,24 @@ func getRootParseNode(responseItem BatchItem) (absser.ParseNode, error) {
 	if contentType == "" {
 		return nil, nil
 	}
-	var buf bytes.Buffer
-	enc := gob.NewEncoder(&buf)
-	err := enc.Encode(responseItem.GetBody())
-	if err != nil {
-		return nil, err
+
+	var (
+		content []byte
+		err     error
+	)
+	if contentType == jsonContentType {
+		if content, err = json.Marshal(responseItem.GetBody()); err != nil {
+			return nil, err
+		}
+	} else {
+		var buf bytes.Buffer
```

Comment:
I wasn't sure if `gob` support is actually needed given that the rest of the code seems to assume JSON. Can remove it if it's cruft.

Reply:
Going with option 1 is good enough here